Content.Shared/_CM14/Xenos/Charge/XenoChargeDoAfterEvent.cs

```diff
@@ -0,0 +1,22 @@
+using Content.Shared.DoAfter;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared._CM14.Xenos.Charge;
+
+[Serializable, NetSerializable]
+public sealed partial class XenoChargeDoAfterEvent : DoAfterEvent
+{
+    [DataField]
+    public NetCoordinates Coordinates;
+
+    public XenoChargeDoAfterEvent(NetCoordinates coordinates)
+    {
+        Coordinates = coordinates;
+    }
+
+    public override DoAfterEvent Clone()
+    {
+        return this;
+    }
+}
```

Comment:
In cases where `Clone()` just returns `this`, you can make the event inherit from `SimpleDoAfterEvent` instead.
```suggestion
public sealed partial class XenoChargeDoAfterEvent : SimpleDoAfterEvent
{
    [DataField]
    public NetCoordinates Coordinates;

    public XenoChargeDoAfterEvent(NetCoordinates coordinates)
    {
        Coordinates = coordinates;
    }
}
```